migrations/2_deploy_admin_contract.js

```diff
@@ -3,6 +3,12 @@ const AllowlistUtils = require('../scripts/allowlist_utils');
 const Admin = artifacts.require("./Admin.sol");
 
 module.exports = async(deployer, network) => {
+    // exit early if we are NOT redeploying this contract
+    let retainCurrentRulesContract = AllowlistUtils.getRetainAdminContract();
+    if (retainCurrentRulesContract) {
+        console.log("not deploying because retain=" + retainCurrentRulesContract);
```

Comment:
I think we could improve the msg specifying the contract that won't be deployed.

e.g.

```
[...] not deploying Admin Contract because [...]
```

Reply:
done